src/org/labkey/targetedms/folderImport/QCFolderImporter.java

```diff
@@ -0,0 +1,110 @@
+    public String getDataType()
+    {
+        return QCFolderConstants.QC_FOLDER_DATA_TYPE;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return QCFolderConstants.QC_FOLDER_DATA_TYPE;
+    }
+
+    @Override
+    public void process(@Nullable PipelineJob job, FolderImportContext ctx, VirtualFile root) throws Exception
+    {
+        if (null != job)
+        {
+            job.setStatus("IMPORT " + getDescription());
+        }
+
+        VirtualFile panoramaQCDir = root.getDir(QCFolderConstants.QC_FOLDER_DIR);
+        List<String> filesToImport = root.getDir(QCFolderConstants.QC_FOLDER_DIR).list();
+        TargetedMSSchema schema = new TargetedMSSchema(ctx.getUser(), ctx.getContainer());
+
+        //iterate through PanoramaQCSettings enum values so that files get imported in that order/ordinal, since the lookup tables need to get populated first
+        for (PanoramaQCSettings qcSetting : PanoramaQCSettings.values())
+        {
+           if (filesToImport.stream().filter(f -> f.equalsIgnoreCase(qcSetting.getSettingsFileName())).count() == 1)
+           {
+               try
+               {
+                   long numRows;
+
+                   if (qcSetting.getSettingsFileName().equalsIgnoreCase(QCFolderConstants.QC_PLOT_SETTINGS_PROPS_FILE_NAME))
+                   {
+                       ctx.getLogger().info("Starting QC Plot settings import");
+                       numRows = qcSetting.importSettingsFromFile(ctx, panoramaQCDir, null, null, null, null);
+                       ctx.getLogger().info("Finished importing " + numRows + " QC Plot settings from " + qcSetting.getSettingsFileName() + " as properties.");
+                   }
+                   else
+                   {
+                       TableInfo ti = schema.getTable(qcSetting.getTableName());
+                       assert ti != null;
+                       QueryUpdateService qus = ti.getUpdateService();
+                       BatchValidationException errors = new BatchValidationException();
+
+                       ctx.getLogger().info("Starting data import from " + qcSetting.getSettingsFileName() + " into targetedms." + qcSetting.getTableName());
+                       numRows = qcSetting.importSettingsFromFile(ctx, panoramaQCDir, schema, ti, qus, errors);
+                       ctx.getLogger().info("Finished importing " + numRows + " rows from " + qcSetting.getSettingsFileName() + " into targetedms." + qcSetting.getTableName());
+                   }
+               }
+               catch (IOException | DuplicateKeyException | BatchValidationException | QueryUpdateServiceException | SQLException e)
+               {
+                   if (qcSetting.getSettingsFileName().equalsIgnoreCase(QCFolderConstants.QC_PLOT_SETTINGS_PROPS_FILE_NAME))
+                   {
+                       LOG.error("Error importing QC Plot settings from " + QCFolderConstants.QC_PLOT_SETTINGS_PROPS_FILE_NAME + ": " + e.getMessage(), e);
```

Comment:
Like with the other errors, this means that nothing will get written to the job's log file and the import will be reported as successful.

Reply:
good point, rethrowing exception now